app/controllers/api/v1/lease_up_applications_controller.rb

```diff
@@ -3,24 +3,57 @@
 module Api::V1
   # Lease Up Applications controller for access via the API
   class LeaseUpApplicationsController < ApiController
+    OFFSET_LIMIT = 2000
+
     def index
       if listing_type == Force::Listing::LISTING_TYPE_FIRST_COME_FIRST_SERVED
         # First Come, First Served listings don't have preferences
         applications = soql_lease_up_application_service.lease_up_applications(lease_up_apps_params)
         applications[:records] = Force::Application.convert_list(applications[:records], :from_salesforce, :to_domain)
+
+        applications[:listing_type] = listing_type
+        render json: applications
       else
         # All other listings need to be queried by preferences first
-        applications = soql_preference_service.app_preferences_for_listing(lease_up_apps_params)
-        applications[:records] = Force::Preference.convert_list(applications[:records], :from_salesforce, :to_domain)
+        application_preferences = get_application_preferences
+        general_applications = get_application_preferences(true)
+        combined = [*application_preferences[:applications], *general_applications[:applications]]
```

Comment:
question: As someone not familiar with this controller, these variable names look like we're mixing data types. Are we combining applications and preferences into a single array?

Reply:
ah wonder if it's because of the naming - I see how it's inconsistent. We are really dealing with application preferences exclusively and appending application data to each application preference, but the list is based on application preferences. In the general case, it's basically a fake application preference for each general application. Does that make sense? Is it worth me going in and updating the naming?

Reply:
Without explicit typing, I would say good variable names are very important in ruby code. Something to keep in mind going forward.